symfony/plugins/orangehrmAdminPlugin/Service/SystemUserService.php

```diff
@@ -0,0 +1,322 @@
+<?php
+/**
+ * OrangeHRM is a comprehensive Human Resource Management (HRM) System that captures
+ * all the essential functionalities required for any enterprise.
+ * Copyright (C) 2006 OrangeHRM Inc., http://www.orangehrm.com
+ *
+ * OrangeHRM is free software; you can redistribute it and/or modify it under the terms of
+ * the GNU General Public License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * OrangeHRM is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this program;
+ * if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA  02110-1301, USA
+ */
+
+namespace OrangeHRM\Admin\Service;
+
+use OrangeHRM\Admin\Dao\SystemUserDao;
+use OrangeHRM\Core\Exception\ServiceException;
+use OrangeHRM\Core\Utility\PasswordHash;
+use OrangeHRM\Entity\SystemUser;
+use OrangeHRM\SecurityAuthentication\Dto\UserCredential;
+
+class SystemUserService {
+    /**
+     * @var SystemUserDao|null
+     */
+    protected ?SystemUserDao $systemUserDao = null;
+    
+    /** @property PasswordHash $passwordHasher */
+    private ?PasswordHash $passwordHasher = null;
+
+    /**
+     * @return SystemUserDao
+     */
+    public function getSystemUserDao(): ?SystemUserDao
+    {
+        if (empty($this->systemUserDao)) {
+            $this->systemUserDao = new SystemUserDao();
+        }
+        return $this->systemUserDao;
+    }
+
+    /**
+     * @param SystemUserDao $systemUserDao
+     */
+    public function setSystemUserDao(SystemUserDao $systemUserDao): void
+    {
+        $this->systemUserDao = $systemUserDao;
+    }
+
+    /**
+     * @return PasswordHash
+     */
+    public function getPasswordHasher(): PasswordHash
+    {
+        if (empty($this->passwordHasher)) {
+            $this->passwordHasher = new PasswordHash();
+        }        
+        return $this->passwordHasher;
+    }
+
+    public function setPasswordHasher(PasswordHash $passwordHasher): void
+    {
+        $this->passwordHasher = $passwordHasher;
+    }
+
+    
+    /**
+     * Save System User
+     * 
+     * @param SystemUser $systemUser
+     * @return void
+     */
+    public function saveSystemUser(SystemUser $systemUser,$changePassword = false){
+        
+        try {
+            if ($changePassword) {
+                $systemUser->setUserPassword($this->hashPassword($systemUser->getUserPassword()));
+            }
+
+            return $this->getSystemUserDao()->saveSystemUser($systemUser);
+            
+        } catch (\Exception $e) {
+            throw new ServiceException($e->getMessage(),$e->getCode(),$e);
```

Comment:
This also not a good way to handle the exception.. we should catch exception in top layers ( Action and APIs)
here should throw an exception if something wrong